src/test/obj_ulog_size/obj_ulog_size.c

```diff
@@ -75,41 +82,55 @@ TOID_DECLARE(struct object, 0);
 #define REDO_OVERFLOW ((size_t)((LANE_REDO_EXTERNAL_SIZE\
 		/ TX_INTENT_LOG_ENTRY_OVERHEAD) + 1))
 
-static int nobj;
-
 /*
  * free_pool -- frees the pool from all allocated objects
+ *		and allocated memory
  */
 static void
-free_pool(PMEMoid *oid)
+free_pool(PMEMoid *oids, size_t noids)
 {
-	for (int i = 0; i < nobj; i++) {
-		pmemobj_free(&oid[i]);
-		UT_ASSERT(OID_IS_NULL(oid[i]));
+	for (int i = 0; i < noids; i++) {
+		pmemobj_free(&oids[i]);
+		UT_ASSERT(OID_IS_NULL(oids[i]));
 	}
+
+	FREE(oids);
 }
 
 /*
- * fill_pool -- fills the pool with maximum amount of objects.
+ * fill_pool -- returns pointer to memory. The memory is being
+ *		dynamically allocated or reallocated when amount of allocated
+ *		memory is too small, and filled with maximum amount of objects.
  *		It tries to fill the pool with object with size of MB.
  *		When it fails, it divides size of the object by DIVIDER,
  *		and tries to fill the pool with object with new smaller size.
  */
-static void
-fill_pool(PMEMobjpool *pop, PMEMoid *oid)
+
+static PMEMoid *
+fill_pool(PMEMobjpool *pop, size_t *noids)
 {
-	nobj = 0;
+	/* 2048 is optimal size to avoid too much reallocations */
+	size_t oids_size = 2048;
+	PMEMoid *oids = MALLOC(oids_size * sizeof(PMEMoid));
+
+	*noids = 0;
 	int ret;
 	/* alloc as much space as possible */
 	for (size_t size = MB; size >= MIN_ALLOC; size /= DIVIDER) {
```

Comment:
What ```MB``` is? A megabyte is the unit not a value or a size.

Reply:
MAX_ALLOC?

Reply:
I think if we realloc with multiplier hardocoded we can also hardcode divider here.